bin/docker-entrypoint.sh

```diff
@@ -0,0 +1,164 @@
+		if [ ! -e .htaccess ]; then
+			# NOTE: The "Indexes" option is disabled in the php:apache base image
+			cat > .htaccess <<-'EOF'
+				# BEGIN WordPress
+				<IfModule mod_rewrite.c>
+				RewriteEngine On
+				RewriteBase /
+				RewriteRule ^index\.php$ - [L]
+				RewriteCond %{REQUEST_FILENAME} !-f
+				RewriteCond %{REQUEST_FILENAME} !-d
+				RewriteRule . /index.php [L]
+				</IfModule>
+				# END WordPress
+			EOF
+			chown www-data:www-data .htaccess
+		fi
+	fi
+
+	# TODO handle WordPress upgrades magically in the same way, but only if wp-includes/version.php's $wp_version is less than /usr/src/wordpress/wp-includes/version.php's $wp_version
+
+	if [ ! -e wp-config.php ]; then
+		awk '/^\/\*.*stop editing.*\*\/$/ && c == 0 { c = 1; system("cat") } { print }' wp-config-sample.php > wp-config.php <<'EOPHP'
+// If we're behind a proxy server and using HTTPS, we need to alert Wordpress of that fact
+// see also http://codex.wordpress.org/Administration_Over_SSL#Using_a_Reverse_Proxy
+if (isset($_SERVER['HTTP_X_FORWARDED_PROTO']) && $_SERVER['HTTP_X_FORWARDED_PROTO'] === 'https') {
+	$_SERVER['HTTPS'] = 'on';
+}
+
+EOPHP
+		chown www-data:www-data wp-config.php
+	fi
+
+	# see http://stackoverflow.com/a/2705678/433558
+	sed_escape_lhs() {
+		echo "$@" | sed 's/[]\/$*.^|[]/\\&/g'
+	}
+	sed_escape_rhs() {
+		echo "$@" | sed 's/[\/&]/\\&/g'
+	}
+	php_escape() {
+		php -r 'var_export((string) $argv[1]);' "$1"
+	}
+	set_config() {
+		key="$1"
+		value="$2"
+		regex="(['\"])$(sed_escape_lhs "$key")\2\s*,"
+		if [ "${key:0:1}" = '$' ]; then
+			regex="^(\s*)$(sed_escape_lhs "$key")\s*="
+		fi
+		sed -ri "s/($regex\s*)(['\"]).*\3/\1$(sed_escape_rhs "$(php_escape "$value")")/" wp-config.php
+	}
+
+	set_config 'DB_HOST' "$WORDPRESS_DB_HOST"
+	set_config 'DB_USER' "$WORDPRESS_DB_USER"
+	set_config 'DB_PASSWORD' "$WORDPRESS_DB_PASSWORD"
+	set_config 'DB_NAME' "$WORDPRESS_DB_NAME"
+
+	# allow any of these "Authentication Unique Keys and Salts." to be specified via
+	# environment variables with a "WORDPRESS_" prefix (ie, "WORDPRESS_AUTH_KEY")
+	UNIQUES=(
+		AUTH_KEY
+		SECURE_AUTH_KEY
+		LOGGED_IN_KEY
+		NONCE_KEY
+		AUTH_SALT
+		SECURE_AUTH_SALT
+		LOGGED_IN_SALT
+		NONCE_SALT
+	)
+	for unique in "${UNIQUES[@]}"; do
+		eval unique_value=\$WORDPRESS_$unique
+		if [ "$unique_value" ]; then
+			set_config "$unique" "$unique_value"
+		else
+			# if not specified, let's generate a random value
+			current_set="$(sed -rn "s/define\((([\'\"])$unique\2\s*,\s*)(['\"])(.*)\3\);/\4/p" wp-config.php)"
+			if [ "$current_set" = 'put your unique phrase here' ]; then
+				set_config "$unique" "$(head -c1M /dev/urandom | sha1sum | cut -d' ' -f1)"
+			fi
+		fi
+	done
+
+	if [ "$WORDPRESS_TABLE_PREFIX" ]; then
+		set_config '$table_prefix' "$WORDPRESS_TABLE_PREFIX"
+	fi
+
+	TERM=dumb php -- "$WORDPRESS_DB_HOST" "$WORDPRESS_DB_USER" "$WORDPRESS_DB_PASSWORD" "$WORDPRESS_DB_NAME" <<'EOPHP'
+<?php
+// database might not exist, so let's try creating it (just to be safe)
```

Comment:
I believe this is better done using WPCLI. See http://wp-cli.org/commands/core/install/